### PR TITLE
Raise unsupported Python version error as an expected error

### PR DESCRIPTION
**Pull request by deivid-rodriguez**

### CODE REVIEW

**Review by jurre — APPROVED** [review]

File: common/lib/dependabot/errors.rb [hunk 1]
```diff
@@ -76,6 +76,20 @@ def initialize(source, msg = nil)
...
+  class ToolVersionNotSupported < DependabotError
+    attr_reader :tool_name, :detected_version, :supported_versions
+
+    def initialize(tool_name, detected_version, supported_versions)
+      @tool_name = tool_name
+      @detected_version = detected_version
+      @supported_versions = supported_versions
+
+      msg = "Dependabot detected the following #{tool_name} requirement for your project: '#{detected_version}'." \
+            "\n\nCurrently, the following Python versions are supported in Dependabot: #{supported_versions}."
```
Comment (abdulapopoola) [inline common/lib/dependabot/errors.rb:88]: should this be `tool` instead of the hardcoded Python?

Comment (deivid-rodriguez) [reply common/lib/dependabot/errors.rb:88]: Oh wow, nice catch! Let me update it!

**Comment by jeffwidman** [comment]
Just curious what the motivation was here as the PR says "as an expected error" but that was already true... ? So this PR instead makes it a new class of error. Is that for internal metrics or greater clarity to the user? 

To be clear, I like the change as `DependencyFileNotResolvable` always felt weird here since it probably_is_resolvable, just not supported by :Dependabot:.

**Comment by deivid-rodriguez** [comment]
Hei! Good to hear from you :)

This error was raised from the fetcher step, not the updater step, and the code in `/updater` does not re-raise `DependencyFileNotResolvable` errors as expected errors (only the updater does). So it was being raised as an internal error instead and this error was showing up in our error tracker instead of being displayed the users in the UI.

**Comment by deivid-rodriguez** [comment]
An alternative fix would've been to add a block [here](https://github.com/dependabot/dependabot-core/blob/22f94ac0044a3db135cc7449bf23b6437dce9647/updater/lib/dependabot/file_fetcher_command.rb#L123-L131) for `DependencyFileNotResolvable` like the one I added for the new error.

But I decided to create a separate error class since it felt cleaner.

**Comment by jeffwidman** [comment]
Hah, that was exactly going to be my follow-up question. Thanks for the clarification!

**Comment by jeffwidman** [comment]
Also, I missed you folks! Last night was the first night I slept longer than four hours at a stretch since the baby arrived, so that's why I finally had the energy to see what's up around these parts.

**Pull request merged**